optimizely/client/client.go

```diff
@@ -133,6 +122,106 @@ func (o *OptimizelyClient) GetEnabledFeatures(userContext entities.UserContext)
 	return enabledFeatures, nil
 }
 
+// GetFeatureVariableString returns string feature variable value
+func (o *OptimizelyClient) GetFeatureVariableString(featureKey string, variableKey string, userContext entities.UserContext) (value string, err error) {
+	val, err := o.getFeatureVariable("string", featureKey, variableKey, userContext)
+	if returnValue, ok := val.(string); ok {
+		value = returnValue
+	}
+	return value, err
+}
+
+func (o *OptimizelyClient) getFeatureVariable(valueType string, featureKey string, variableKey string, userContext entities.UserContext) (value interface{}, err error) {
```

Comment:
I think conceptually it will be better if that function returns the string instead of interface and GetFeatureVariableString does the checking. That way we eliminate valueType parameter.

Reply:
actually GetFeatureVariableString will not have to do any checking - just pass-through, but GetFeatureVariableBool, GetFeatureVariableFloat, etc  will do some checking

Reply:
That is a great point. I think we should do it that way even if it ends up duplicating some of the code at the bottom since strong typing does give us other kinds of guarantees @yasirfolio3

Reply:
@pawels-optimizely we need valueType parameter for this condition
https://github.com/optimizely/go-sdk/pull/71/files#diff-b852811f33e54bd7b6e89917e41eefadR203

Reply:
@mikeng13 yes we move the checking part to each individual api such as GetFeatureVariableString, but we still do need valueType argument to check if variable.Type is the required valueType, any thoughts?

Reply:
my understanding is that you will have GetFeatureVariableString, GetFeatureVariableBool, GetFeatureVariableFloat - then they can check (return error) if a string cannot be converted to a specific type. Unless I am missing something...

Reply:
> my understanding is that you will have GetFeatureVariableString, GetFeatureVariableBool, GetFeatureVariableFloat - then they can check (return error) if a string cannot be converted to a specific type. Unless I am missing something...

yes that's correct, what about the valueType parameter? we would still need it for this condition
https://github.com/optimizely/go-sdk/pull/71/files#diff-b852811f33e54bd7b6e89917e41eefadR203

Reply:
> I think conceptually it will be better if that function returns the string instead of interface and GetFeatureVariableString does the checking. That way we eliminate valueType parameter.

pointing to your suggestion about elimination valueType parameter.

Reply:
that can be checked by individual function: GetFeatureVariableBool will check for bool, GetFeatureVariableFloat will check for float, etc. All you care about is whether you get an error or not when you try to convert to a particular type.

Reply:
what about GetFeatureVariableString? since variable.Value is a string type, it will find any value as a valid string and return it. for e.g:
if variable.Value = "2" and variable.Type="integer", then GetFeatureVariableString should be returning an error. But since we are talking about removing valueType parameter, we will also have to remove the condition value.Type == valueType, which will cause GetFeatureVariableString to return any kind of value as a valid string.

Reply:
you are calling it as getFeatureVariable("string", featureKey, variableKey, userContext), that means variable "2" would be treated as a string in GetFeatureVariableString

Reply:
also if you need the type, you can return a string and the type and check that in GetFeatureVariableString

Reply:
> also if you need the type, you can return a string and the type and check that in GetFeatureVariableString

il go ahead with this idea!

Reply:
yeah, sorry to be a pain - I am not 100% on this code yet, but it's kind of awkward flow when I read the code: 1. getting a string, 2. converting to interface, 3. returning the interface, 4. converting interface to string again - this is your overall flow. Just trying to avoid that.

Reply:
no worries, can you review it now!